File: covariance.py
# for vars x & y, fill these w/your data

x = []
y = []
x_deviation = []
y_deviation = []
x_mean = 12.56
y_mean = 2.04

## Variance Section

# finding variance for x
#If you're attempting to find covariance, comment out the deviation = deviation ** 2 part on both the x loop & y loop
for i in x:
	deviation = i - x_mean
	deviation = deviation ** 2
	x_deviation.append(deviation)

#prints the sum for your x variance, then you divide by N-1
z = sum(x_deviation)
print(z)

# finding variance for y
for i in y:
	deviation = i - y_mean
	deviation = deviation ** 2
	y_deviation.append(deviation)

#prints the sum for your y variance, then you divide by N-1
z = sum(y_deviation)
print(z)


## Covariance
multiplied_num = []
#Make sure to comment out the deviation**2 parts of the code above if you're attempting to find the covariance
# Change the range int according to the length of the X & Y values, this will iterate through the multiplication part 
# of the equation 25 times in order to multiply every number, then it adds them to the Z variable
#for i in range(25):
#	number = x_deviation[i] * y_deviation[i]
#	multiplied_num.append(number)

# prints sum for your x & y covariance, then you divide by N-1
#z = sum(multiplied_num)
#print(z)

### by the population size (N) - 1

File: covariance.c
#include <stdio.h>

int main()
{
	int x[] = {20, 22, 4.4};
	int y[] = {78, 73, 71};
	int x_deviation[3];
	int y_deviation[3];
	float x_mean = 15.5;
	float y_mean = 74;
	float deviationX = 0.0;
	float deviationY = 0.0;
	int sum;
	int multiplied_nums[3];
	int n;
	FILE *f;
	f = fopen("/home/n9/calculations.txt", "w+");
	fprintf(f, "NOTE: THESE VALUES AREN'T 100 PERCENT ACCURATE AND SHOULD BE TREATED AS ESTIMATES\n\n");
	/* Calculating covariance for X */
	for ( int a=0; a < sizeof(x)/sizeof(x[0]); a++ ) 
	{
		deviationX = x[a] - x_mean;
		x_deviation[a] = deviationX;
		printf("X deviation: %d\n", x_deviation[a]);
		/* outputs X deviations to a file */
		fprintf(f, "X Deviation: %d\n", x_deviation[a]);
	}
	/* Calculating covariance for Y */
	for ( int a=0; a < sizeof(y)/sizeof(y[0]); a++ )
	{
		deviationY = y[a] - y_mean;
		y_deviation[a] = deviationY;
		printf("Y deviation: %d\n", y_deviation[a]);
		fprintf(f, "Y Deviation: %d\n", y_deviation[a]);
	}
	/* Change value of 3 depending on how large array of numbers is */
	/* This will loop through each X number & multiply it by its corresponding Y number*/
	for ( int loop=0; loop < 3; loop++ )
	{
		multiplied_nums[loop] = x_deviation[loop] * y_deviation[loop];
	}
	/* This will loop through the array of multiplied_nums and add them all up*/
	sum = 0;
	for ( int loop=0; loop < sizeof(multiplied_nums)/sizeof(multiplied_nums[0]); loop++) 
	{
		sum = sum + multiplied_nums[loop];
	}
	printf("The sum of multiplying the values & adding together the sum : %d\n", sum);
	fprintf(f, "Sum of multiplying values & adding them together: %d\n", sum);
	n = sizeof(multiplied_nums)/sizeof(multiplied_nums[0]);
	n = n - 1;
	printf("The covariance for X & Y is: %d\n", sum/n);
	fprintf(f, "Covariance of X & Y is estimated to be: %d\n", sum/n);
		
		

}
